mern-ecommerce/controllers/auth.js

```diff
@@ -62,20 +62,20 @@ exports.requireSignin = expressJwt({
   userProperty: 'auth',
 });
 
-exports.isAuth = (req, res, next) => {
-  let user = req.profile && req.auth && req.profile._id == req.auth._id;
-  if (!user) {
+exports.isAdmin = (req, res, next) => {
+  if (req.profile.role === 0) {
     return res.status(403).json({
-      error: 'Access denied',
+      error: 'Admin resource! Access denied',
     });
   }
   next();
 };
 
-exports.isAdmin = (req, res, next) => {
-  if (req.profile.role === 0) {
+exports.isAuth = (req, res, next) => {
```

Comment:
The `isAuth` middleware function seems to be checking if the user is authenticated by verifying the `_id` of the `req.profile` and `req.auth`. However, there is no validation or check to ensure that `req.profile` and `req.auth` exist before trying to access their `_id` properties. This could lead to a runtime error if either of these properties is undefined.

Additionally, it would be better to use a more explicit check for the existence of `req.profile` and `req.auth` before trying to access their properties. This can be done using the `&&` operator or the optional chaining operator (`?.`) introduced in JavaScript.

For example:
```
let user = req.profile?.&_id === req.auth?.&_id;
```
This will prevent any runtime errors and make the code more robust.